src/bitflags_trait.rs

```diff
@@ -0,0 +1,68 @@
+use core as _core;
+
+pub trait BitFlags<T>:
```

Comment:
We should probably make this an associated type, since an implementation of `BitFlags` only corresponds to a single underlying integer type.

Reply:
I think the lines in this comment are wrong? do you mean perhaps empty and all?

Reply:
Ah sorry, I mean making the trait look like:

```rust
pub trait BitFlags {
    type Bits;

    ..
}
```

instead of:

```rust
pub trait BitFlags<T> {
    ..
}
```